game-core/src/main/java/games/strategy/debug/console/window/ConsoleModel.java

```diff
@@ -0,0 +1,109 @@
+package games.strategy.debug.console.window;
+
+import java.util.Collection;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import games.strategy.triplea.settings.ClientSetting;
+import games.strategy.util.SystemClipboard;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.java.Log;
+
+@Log
+class ConsoleModel {
+
+  private final ConsoleView consoleView;
+
+  ConsoleModel(final ConsoleView consoleView) {
+    this.consoleView = consoleView;
+
+    // Show console window automatically when the user setting is toggled on.
+    ClientSetting.showConsole.addListener(gameSetting -> {
+      if (gameSetting.getValueOrThrow()) {
+        consoleView.setVisible(true);
+      }
+    });
+
+    // if the console is closed by user manually, keep it closed and do not show it again on startup.
+    consoleView.addWindowClosedListener(() -> ClientSetting.showConsole.setValueAndFlush(false));
+  }
+
+  static String getCurrentLogLevel() {
+    return ClientSetting.loggingVerbosity.getValue()
+        .map(Level::parse)
+        .map(LogLevelItem::fromLevel)
+        .orElse(LogLevelItem.NORMAL.label);
+  }
+
+  static Collection<String> getLogLevelOptions() {
+    return Stream.of(LogLevelItem.values())
+        .map(logLevelSelection -> logLevelSelection.label)
+        .collect(Collectors.toList());
+  }
+
+  void memoryAction() {
+    consoleView.append(DebugUtils.getMemory());
+  }
+
+  void propertiesAction() {
+    consoleView.append(DebugUtils.getProperties());
+  }
+
+  void copyToClipboardAction() {
+    SystemClipboard.setClipboardContents(consoleView.readText());
```

Comment:
Instead of having this class provide static methods, why don't you just create another testing-only constructor overload that accepts an instance of `SystemClipboard` (defaulting to the current one) that can easily get mocked in tests without having to re-set the actual variable.

Reply:
testing-only constructors have some virtue but I've generally come to find that they are not the best pattern. I think it's always generally understood that `@VisibileForTesting` is a compromise. 

I don't think there is necessarily a lot of value to mock verify that we call untested code, may as well just call the untested code.

OTOH, I am interested in solving the design problem of how view-model should use strategy objects.  It was as bit of a larger update to get there: 50a2e246a800af79530db03d8a9a2a85bc705461

Two notable updates from that commit:
>    1. break console model and view circular dependency:
> 2. Inject a strategy object for setting system clipboard into console
    model.